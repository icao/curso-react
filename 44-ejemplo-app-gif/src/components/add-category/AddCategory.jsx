import React, { useState } from 'react'

const AddCategory = ({ setCategorias }) => {
  const [categoria, setCategoria] = useState('')

  function handleChange(event) {
    let value = event.target.value
    setCategoria(value)
  }

  function handleSubmit(event) {
    event.preventDefault()
    console.log(`Agregando... üì™ ${categoria}`)
    // props.setCategorias(categorias => [...categorias, categoria]) // [1]

    // Validando la longitud de la entrada en el input
    if (categoria.trim().length > 2) {
      setCategorias(categorias => [...categorias, categoria]) // [2]
      setCategoria('')
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor='imput-category'>
        Categor√≠a{' '}
        <input
          id='imput-category'
          type='text'
          value={categoria}
          onChange={handleChange}
        />
      </label>
      <button>Agregar</button>
    </form>
  )
}
export default AddCategory

/**
 * [1] Si no destructuramos las props as√≠ se llamar√≠a la funsi√≥n
 *
 * [2]  No disponemos del arreglo de categor√≠as, por consiguiente, usamos un callback en el cual esperamos las categor√≠as del estado anterior y regresamos un arreglo con el anterior estado y la nueva categoria.
 * Si se ubieran proporcionado las categor√≠as, la actualizaci√≥n ser√≠a la sig: setCategorias([...categorias, categoria])
 *
 */
