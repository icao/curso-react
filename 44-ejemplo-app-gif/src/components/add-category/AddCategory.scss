@import '../../scss/Palette';

.form-container {
  width: 100%;
  height: auto;
  display: flex;
  justify-content: center;
  align-items: center;
}

.input-category {
  color: $color-grey-regular;
  border: none;
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  width: 100%;
  height: 50px;
  font-size: 1.3rem;
  padding: 1rem;
  outline: none;
}

.btn {
  height: 100%;
  padding: 1rem;
  font-size: 1.1rem;
  border: none;
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: none;
}

.btn-primary {
  height: 50px;
  background: linear-gradient(
    45deg,
    $color-fiucsia 14%,
    $color-pink-mexican 84%
  );
  color: $color-white;
}
