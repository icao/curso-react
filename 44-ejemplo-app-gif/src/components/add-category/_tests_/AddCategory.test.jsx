import AddCategory from '../AddCategory'
import { shallow } from 'enzyme'

describe('Probando el componente <AddCategory />', () => {
  test('El componente <AddCategory /> debe de mostrarse correctamente y aceptar la prop setCategorias', () => {
    // NOTA: Por el momento inventaremos una función dummy, en otros capítulos futuros, verémos la correcta forma de enviarle una función como parametro
    const setCategorias = () => {} // Esta función dummy no hace nada
    const wrapper = shallow(<AddCategory setCategorias={setCategorias} />)
    expect(wrapper).toMatchSnapshot()
  })
})
