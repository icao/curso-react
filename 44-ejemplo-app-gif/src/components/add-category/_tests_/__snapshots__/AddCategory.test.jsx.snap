// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <AddCategory /> El componente <AddCategory /> debe de mostrarse correctamente y aceptar la prop setCategorias 1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <input
    className="input-category"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn-primary"
  >
    ðŸ”Ž
  </button>
</form>
`;
